src/transformers/models/levit/modeling_tf_levit.py

```diff
@@ -0,0 +1,1048 @@
+    # See all LeViT models at https://huggingface.co/models?filter=levit
+]
+
+
+@dataclass
+class TFLevitForImageClassificationWithTeacherOutput(ModelOutput):
+    """
+    Output type of [`TFLevitForImageClassificationWithTeacher`].
+
+    Args:
+        logits (`tf.Tensor` of shape `(batch_size, config.num_labels)`):
+            Prediction scores as the average of the `cls_logits` and `distillation_logits`.
+        cls_logits (`tf.Tensor` of shape `(batch_size, config.num_labels)`):
+            Prediction scores of the classification head (i.e. the linear layer on top of the final hidden state of the
+            class token).
+        distillation_logits (`tf.Tensor` of shape `(batch_size, config.num_labels)`):
+            Prediction scores of the distillation head (i.e. the linear layer on top of the final hidden state of the
+            distillation token).
+        hidden_states (`tuple(tf.Tensor)`, *optional*, returned when `output_hidden_states=True` is passed or when `config.output_hidden_states=True`):
+            Tuple of `tf.Tensor` (one for the output of the embeddings + one for the output of each layer) of shape
+            `(batch_size, sequence_length, hidden_size)`. Hidden-states of the model at the output of each layer plus
+            the initial embedding outputs.
+    """
+
+    logits: tf.Tensor = None
+    cls_logits: tf.Tensor = None
+    distillation_logits: tf.Tensor = None
+    hidden_states: Optional[Tuple[tf.Tensor]] = None
+
+
+class TFLevitConvEmbeddings(tf.keras.layers.Layer):
+    """
+    LeViT Conv Embeddings with Batch Norm, used in the initial patch embedding layer.
+    """
+
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        kernel_size,
+        stride,
+        padding,
+        dilation=1,
+        groups=1,
+        bn_weight_init=1,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        # The padding layer is built in order to pad the inputs before entering the convolution operation.
+        self.padding = tf.keras.layers.ZeroPadding2D(padding=padding)
+        self.convolution = tf.keras.layers.Conv2D(
+            filters=out_channels,
+            kernel_size=kernel_size,
+            strides=stride,
+            dilation_rate=dilation,
+            groups=groups,
+            use_bias=False,
+            data_format="channels_last",
+            name="convolution",
+        )
+        # The epsilon and momentum used here are the defaults in torch batch norm layer.
+        self.batch_norm = tf.keras.layers.BatchNormalization(epsilon=1e-05, momentum=0.9, name="batch_norm")
+
+    def call(self, embeddings: tf.Tensor, training: Optional[bool] = None):
+        # embeddings shape = (bsz, num_channels, height, width)
+        embeddings = tf.transpose(embeddings, perm=(0, 2, 3, 1))
+        embeddings = self.padding(embeddings)
+        embeddings = self.convolution(embeddings, training=training)
```

Comment:
I am assuming you've verified if this is indeed the sequence to follow when matching a `torch.nn.Conv2d()` with `padding` defined?

Reply:
https://github.com/huggingface/transformers/blob/7032e0203262ebb2ebf55da8d2e01f873973e835/src/transformers/models/regnet/modeling_tf_regnet.py#L66

I have actually used this comment and colab notebook linked.